storage/memory/storage.go

```diff
@@ -227,3 +230,13 @@ func (s *ShallowStorage) SetShallow(commits []plumbing.Hash) error {
 func (s ShallowStorage) Shallow() ([]plumbing.Hash, error) {
 	return s, nil
 }
+
+type ModuleStorage map[string]*Storage
+
+func (s ModuleStorage) Module(name string) (storage.Storer, error) {
+	if _, ok := s[name]; !ok {
+		s[name] = NewStorage()
+	}
+
+	return s[name], nil
```

Comment:
Don't look up twice in the map, use a variable to store the results of the first lookup.